### PR TITLE
mastercontainer - limit access to php-fpm to localhost

### PR DESCRIPTION
**Pull request by szaimen**
address part of https://github.com/nextcloud/all-in-one/issues/3172

### CODE REVIEW

File: Containers/mastercontainer/Dockerfile [hunk 1]
```diff
@@ -56,6 +56,8 @@ RUN set -ex; \
     sed -i 's/^pm = dynamic/pm = ondemand/' /usr/local/etc/php-fpm.d/www.conf; \
     sed -i 's/^pm.max_children =.*/pm.max_children = 80/' /usr/local/etc/php-fpm.d/www.conf; \
     sed -i 's|access.log = /proc/self/fd/2|access.log = /proc/self/fd/1|' /usr/local/etc/php-fpm.d/docker.conf; \
+    grep -q ';listen.allowed_clients' /usr/local/etc/php-fpm.d/www.conf; \
+    sed -i 's|;listen.allowed_clients.*|listen.allowed_clients = 127.0.0.1,::1|' /usr/local/etc/php-fpm.d/www.conf; \
```
Comment (Zoey2936) [inline Containers/mastercontainer/Dockerfile:60]:
```suggestion
    grep -q 'listen =' /usr/local/etc/php-fpm.d/www.conf; \
    sed -i 's|listen =.*|listen = /var/run/php.sock|' /usr/local/etc/php-fpm.d/www.conf; \
```

And change the Apache PHP handler to: `SetHandler "proxy:unix:/var/run/php.sock"`

**Comment by Zoey2936** [comment]
also not needed, since php fpm is by default only exposed to 127.0.0.1 if I remember correctly

**Comment by Zoey2936** [comment]
```console
/var/www/html # cat /usr/local/etc/php-fpm.d/www.conf | grep listen | grep -v ";"
listen = 127.0.0.1:9000
```

**Comment by szaimen** [comment]
> ```
> /var/www/html # cat /usr/local/etc/php-fpm.d/www.conf | grep listen | grep -v ";"
> listen = 127.0.0.1:9000
> ```

It is the same in the nextcloud container and there access from a different container is obiously possible so switching to a socket probably makes sense...

File: Containers/mastercontainer/Dockerfile (hunk 1 above)
Comment (szaimen) [reply Containers/mastercontainer/Dockerfile:60]: all right, lets do it like this then

Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:60]: https://github.com/docker-library/php/blob/e9e47fd185d6b36e491c713811437b0c17944faa/8.2/alpine3.18/fpm/Dockerfile#L249

```suggestion
    grep -q 'listen =' /usr/local/etc/php-fpm.d/www.conf; \
    sed -i 's|listen =.*|;listen = /var/run/php.sock # handled in zz-docker.conf|' /usr/local/etc/php-fpm.d/www.conf; \
    grep -q 'listen =' /usr/local/etc/php-fpm.d/zz-docker.conf; \
    sed -i 's|listen =.*|listen = /var/run/php.sock|' /usr/local/etc/php-fpm.d/zz-docker.conf; \
```

And change the Apache PHP handler to: `SetHandler "proxy:unix:/var/run/php.sock"`

Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:60]: healthcheck also needs to check if /var/run/php.sock exists

Comment (szaimen) [reply Containers/mastercontainer/Dockerfile:60]: or we just actually limit it to 127.0.0.1:::1? would that be fine for you? Sounds like the easier approach...

Comment (szaimen) [reply Containers/mastercontainer/Dockerfile:60]: due to the performance improvement?

Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:60]: https://serverfault.com/questions/124517/what-is-the-difference-between-unix-sockets-and-tcp-ip-sockets unix sockets also seems to be faster

Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:60]: I don't see why it is simpler to limit ips, instead of using a socket

Comment (szaimen) [reply Containers/mastercontainer/Dockerfile:60]: 2 lines of code vs 4? but yeah if you really want we can also use a socket

Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:60]: I would use the socket

Comment (szaimen) [reply Containers/mastercontainer/Dockerfile:60]: @Zoey2936 lets switch to the socket in a follow-up.

I would welcome a PR :)

**Comment by szaimen** [comment]
This is now released with v7.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**